apps/supaclk/app.js

```diff
@@ -0,0 +1,195 @@
+    32,
+    atob("BAUHCwsNEQQGBgkKBQkFBwwHCgkKCQoJCgoFBQsLCwkQDAwNDgwLDQ8HBw0LEQ0PDA8NCwwMDBINCwsGBwY="),
+    18|65536
+  );
+}
+
+// Actually draw the watch face
+let draw = function() {
+  g.reset().setColor(g.theme.fg).setBgColor(g.theme.bg);
+  g.clearRect(R.x,R.y, R.x2, R.y2 - 60);
+  var date = new Date();
+  let minutes = date.getMinutes();
+  minutes = (minutes < 10 ? '0' : '') + minutes;
+
+  g.setColor(g.theme.fg)
+  // Time
+  const yt =  R.y + 92 - 20 - 30 + 6 + 10;
+  const xt = R.w/2 - 5;
+  let hours = date.getHours()+'';
+  g.setFontAlign(1, 0).setFontPlayfairDisplay().drawString(hours, xt - 8, yt);
+  g.setFontAlign(0, 0).setFontPlayfairDisplay().drawString(':', xt, yt);
+  g.setFontAlign(-1, 0).setFontPlayfairDisplay().drawString(minutes, xt + 8, yt);
+  // logo
+  g.drawImage(supaClockImg, R.x2 - supaClockImg.width - 2, R.y + 2);
+  // dow + date
+  let dateStr = require("locale").dow(date).toUpperCase() + '\n' + require("locale").month(date, 2).toUpperCase() + ' ' + date.getDate();
+  g.setFont('6x8').setFontAlign(-1, 0).drawString(dateStr, R.x2 - supaClockImg.width - 2, R.y + 42 - 30 + 8);
+
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(function() {
+    drawTimeout = undefined;
+    draw();
+  }, 60000 - (Date.now() % 60000));
+};
+
+let clockInfoDraw = (itm, info, options) => {
+  g.reset().setFontPlayfairDisplaySm().setColor(g.theme.fg).setBgColor(g.theme.bg);
+  if (options.focus) g.setBgColor("#FF0");
+  g.clearRect({x:options.x,y:options.y,w:options.w,h:options.h});
+
+  if (info.img) {
+    g.drawImage(info.img, options.x+1,options.y+2);
+  }
+  var text = info.text.toString().toUpperCase();
+  if (g.setFontPlayfairDisplaySm().stringWidth(text)+24-2>options.w) g.setFont("8x12");
+  g.setFontAlign(-1,-1).drawString(text, options.x+24+3, options.y+6);
+};
+
+let clockInfoDrawR = (itm, info, options) => {
+  g.reset().setFontPlayfairDisplaySm().setColor(g.theme.fg).setBgColor(g.theme.bg);
+  if (options.focus) g.setBgColor("#FF0");
+  g.clearRect({x:options.x,y:options.y,w:options.w,h:options.h});
+
+  if (info.img) {
+    g.drawImage(info.img, options.x + options.w-1-24,options.y+2);
+  }
+  var text = info.text.toString().toUpperCase();
+  if (g.setFontPlayfairDisplaySm().stringWidth(text)+24-2>options.w) g.setFont("8x12");
+  g.setFontAlign(1,-1).drawString(text, options.x+options.w-24-3, options.y+6);
+};
+
+
+let clockInfoItems;
+let clockInfoMenu1;
+let clockInfoMenu2;
+let clockInfoMenu3;
+let clockInfoMenu4;
+
+// Show launcher when middle button pressed
+Bangle.setUI({
+  mode : "clock",
+  remove : function() {  // for fastloading, clear the app memory
+    if (drawTimeout) clearTimeout(drawTimeout);
+    drawTimeout = undefined;
+    delete Graphics.prototype.setFontPlayfairDisplay
+    delete Graphics.prototype.setFontPlayfairDisplaySm
+    delete clockInfoItems;
+    clockInfoMenu1.remove();
```

Comment:
If I press the HW button when the splash screen in showing the screen will be blank afterwards with this error message in the Web IDE, on fw 2v22.5:

```
 ____                 _
|  __|___ ___ ___ _ _|_|___ ___
|  __|_ -| . |  _| | | |   | . |
|____|___|  _|_| |___|_|_|_|___|
         |_| espruino.com
 2v22.5 (c) 2024 G.Williams
>
Uncaught Error: Can't read property 'remove' of undefined
 at line 155 col 19 in supaclk.app.js
    clockInfoMenu1.remove();
                  ^
in function "a" called from line 2 col 239
...e;delete Bangle.uiRemove;a()}g.reset();if(c){if("updown"==c)...
                              ^
in function called from line 25 col 338 in .boot0
...options.back;}setUI(mode,cb);
                              ^
at line 25 col 338 in .boot0
...options.back;}setUI(mode,cb);
                              ^
in function "setUI" called from line 1 col 31
Bangle.uiRemove?(Bangle.setUI(),(__FILE__=a)||(a=".bootcde")...
                              ^
in function called from line 15 col 1797 in .boot0
...);Bangle.load=orig;}else o(name);
                                  ^
at line 15 col 1797 in .boot0
...);Bangle.load=orig;}else o(name);
                                  ^
in function "load" called from line 1 col 408
...))}c?Bangle.load(b.launcher):eval('E.showMessage("No Launche...
                              ^
in function called from system
Uncaught Error: Unhandled promise rejection: Error: Function "setFontPlayfairDisplay" not found!
 at line 97 col 24 in supaclk.app.js
  g.setFontAlign(1, 0).setFontPlayfairDisplay().drawString(h...
                       ^
in function "draw" called from line 175 col 8 in supaclk.app.js
  draw();
       ^
in function called from system
> 
```
I tried this suggestion

```suggestion
    clockInfoMenu1&&clockInfoMenu1.remove();
```

... but it didn't play well. The splash screen would continue to draw over the loaded launcher and eventually clear the screen. The launcher still works though if I swipe the screen to interact. Maybe the splash screen could be made to stop short if the HW button is pressed? Maybe not for this PR if you don't want to. Anyway I subsequently also got this error message with the modified code:

```
 ____                 _
|  __|___ ___ ___ _ _|_|___ ___
|  __|_ -| . |  _| | | |   | . |
|____|___|  _|_| |___|_|_|_|___|
         |_| espruino.com
 2v22.5 (c) 2024 G.Williams
>
Uncaught Error: Unhandled promise rejection: Error: Function "setFontPlayfairDisplay" not found!
 at line 97 col 24 in supaclk.app.js
  g.setFontAlign(1, 0).setFontPlayfairDisplay().drawString(h...
                       ^
in function "draw" called from line 175 col 8 in supaclk.app.js
  draw();
       ^
in function called from system
Uncaught Error: Unhandled promise rejection: Error: Function "setFontPlayfairDisplay" not found!
 at line 97 col 24 in supaclk.app.js
  g.setFontAlign(1, 0).setFontPlayfairDisplay().drawString(h...
                       ^
in function "draw" called from line 175 col 8 in supaclk.app.js
  draw();
       ^
in function called from system
>
```

In conclusion, I don't have a short and neat solution to suggest. Maybe you could try catching the original error for now and do something if it occurs?

Reply:
Maybe do a check for if `setFontPlayfairDisplay` is defined and if it's not `return`?

Reply:
Ok, I took a dig at this and think I have solved it. I'll do a PR to your fork/branch if you want to merge it and have it go in with this PR to the official repo.